sdk/python/feast/registry.py

```diff
@@ -537,3 +542,80 @@ def _write_registry(self, registry_proto: RegistryProto):
         file_obj.seek(0)
         blob.upload_from_file(file_obj)
         return
+
+
+class S3RegistryStore(RegistryStore):
+    def __init__(self, uri: str):
+        self._uri = urlparse(uri)
+        self._bucket = self._uri.hostname
+        self._key = self._uri.path.lstrip("/")
+        try:
+            import boto3
+        except ImportError as e:
+            from feast.errors import FeastExtrasDependencyImportError
+
+            raise FeastExtrasDependencyImportError("aws", str(e))
+        self.s3 = boto3.resource(
+            "s3", endpoint_url=os.environ.get("FEAST_S3_ENDPOINT_URL")
+        )
+        return
+
+    def get_registry_proto(self):
+        file_obj = TemporaryFile()
+        registry_proto = RegistryProto()
+        try:
+            import botocore
```

Comment:
Let's do the try import of botocore in the __init__ together with boto3. Earlier we raise the exception on this, the better. Then here, you'll just be able to write `import botocore` without try/catch.

Reply:
This comment still stands ^

Reply:
still the same answer
```
feast/registry.py:551:13: F401 'botocore' imported but unused
feast/registry.py:570:16: F821 undefined name 'botocore'
feast/registry.py:585:16: F821 undefined name 'botocore'
make: *** [lint-python] Error 1
```

Reply:
I see, that makes sense then, forgot this comment.